.github/workflows/tests-typecheck.yml

```diff
@@ -0,0 +1,63 @@
+name: Tests, Linter & Typecheck
+
+on:  
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```

Comment:
I think we should run this on all pull requests, not just the ones to `main` (It'll be most commonly used with main of course, but we should have it on for all anyway)

Reply:
100%